runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -155,7 +155,17 @@ bool TR_EscapeAnalysis::isImmutableObject(TR::Node *node)
    if (node->getOpCodeValue() != TR::New)
       return false;
 
-   if (!strncmp("java/lang/Integer", getClassName(node->getFirstChild()), 17))
+   char *className = getClassName(node->getFirstChild());
```

Comment:
getClassName may not be new, but I think there needs to be a safety check that the node is one which should have a class as its symbol and that we can get the name. We should handle this returning NULL in cases where it can't be gotten.

Reply:
Thanks, Andrew @andrewcraik.  I've been pawing around trying to figure out the right way to perform the check you're recommending.  Would this be the right way?

```
if (classNode->getOpCode().hasSymbolReference())
   {
   TR::SymbolReference *symRef = classNode->getSymbolReference();
   TR::Symbol *sym = symRef->getSymbol();
   if (sym->isClassObject() && !sym->addressIsCPIndexOfStatic())
      ...
   }
```

Reply:
So the child node will need to be a loadaddr otherwise the symref won't be describing the thing we want (currently news have a loadaddr as a child, but this will be changing). So the first part of the check needs to be that we have a loadaddr.

Usage elsewhere in the compiler suggests sym->isClassObject() is the check you need. I don't know that you need the addressIsCPIndexOfStatic.

Reply:
Added checking to `getClassName` and its callers in commit f12434d607be9a31b4674ea1c8bcedceb82558da